src/lrs_annotation/utils.py

```diff
@@ -0,0 +1,381 @@
+"""
+Utility methods used across the workflows
+"""
+import hashlib
+from hailtop.batch.job import BashJob
+from enum import Enum
+from cpg_utils import Path
+from cpg_utils.cloud import read_secret
+from cpg_utils.config import ConfigError, config_retrieve, reference_path, image_path
+from os.path import join
+from random import randint
+from typing import Any
+
+from functools import cache
+
+from hailtop.batch.job import Job
+
+from cpg_utils.cromwell import CromwellOutputType, run_cromwell_workflow_from_repo_and_get_outputs
+from cpg_utils.hail_batch import command, get_batch
+
+GATK_SV_COMMIT = 'dc145a52f76a6f425ac3f481171040e78c0cfeea'
+
+
+class CromwellJobSizes(Enum):
+    """
+    Enum for polling intervals
+    """
+
+    SMALL = 'small'
+    MEDIUM = 'medium'
+    LARGE = 'large'
+
+
+def get_dataset_name(dataset: str) -> str:
+    """
+    Add -test suffix to dataset name if in test mode.
+    """
+    test = config_retrieve(['workflow', 'access_level']) == 'test'
+    return dataset + '-test' if test else dataset
+
+
+def get_dataset_names(datasets: str | list[str]) -> list[str]:
+    """
+    Add -test suffix to dataset names if in test mode.
+    """
+    return [get_dataset_name(dataset) for dataset in datasets]
+
+
+def get_query_filter_from_config(field_name: str, make_tuple = True) -> tuple[str] | list[str] | None:
+    """
+    Get values for the specified field from the workflow.lrs_annotation config dictionary.
+
+    Returns tuples by default, because they are needed for use with cached functions.
+    """
+    if values := config_retrieve(
+        ['workflow', 'query_filter', field_name],
+        default=None,
+    ):
+        if isinstance(values, str):
+            values = (values,)
+        if not make_tuple:
+            return list(values)
+        return tuple(values)
+    return None
+
+
+def get_sg_hash(sequencing_group_ids: list[str]) -> str:
+    """
+    Unique hash string of sequencing group IDs
+    Reimplemented from CPG flow core for use with LRS annotation pipeline, which sometimes
+    uses subsets of sequencing groups in a given multicohort.
```

Comment:
interesting - for a given Cohort isn't it still true that we want to process all SGs within that Cohort?